### DISCUSSION TITLE
Modal Dialog

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#237] [Component suggestions]
<img width="720" alt="visuele weergave van het modal dialog component" src="https://github.com/nl-design-system/backlog/assets/545207/ae9d3fc2-d274-4595-ad46-f845e06d419e">

We willen graag meer leren over het **Modal Dialog** component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1712323790754919)

---

## Naam
Modal Dialog

## Doel
Toont een venster over de pagina om de aandacht op een enkele taak of bericht te vestigen. Interactie met onderliggende pagina wordt geblokkeerd.

## Gerelateerde componenten
[Alert Dialog](https://github.com/orgs/nl-design-system/discussions/302), [Drawer](https://github.com/orgs/nl-design-system/discussions/238), [Dialog](https://github.com/orgs/nl-design-system/discussions/280)

## Onderzoek
[Analyse van inzichten opgehaald tijdens gebruikersonderzoek bij gemeente Den Haag - Miro](https://miro.com/app/board/uXjVNA__E6Y=/?moveToWidget=3458764574615381616&cot=14)
Miro board waar inzichten uit een gebruikersonderzoek van de gemeente Den Haag zijn beschreven. Tijdens het onderzoek werd er een [prototype van een meerstappenformulier](https://www.figma.com/proto/iXcHSzJFHwCsdvPVdcakk2/%F0%9F%93%8B-Den-Haag-formulieren?page-id=489%3A57853&type=design&node-id=490-54675&viewport=720%2C736%2C0.04&t=rRsxTTMBYhCVkuPQ-1&scaling=min-zoom&mode=design) voorgelegd aan ondernemers. Hierbij werd onder andere het patroon ‘Opslaan en later verdergaan’ doorlopen waar het Modal Dialog component voor werd ingezet. Inzichten: CTA in modal valt op en men gaat er goed mee overweg.

[Dialogs and popovers seem similar. How are they different? - Hidde de Vries](https://hidde.blog/dialog-modal-popover-differences/)
Blogpost waarin Hidde de verschillen tussen Dialogs en pop-overs uiteen zet. 

[Modal & Nonmodal Dialogs: When (& When Not) to Use Them - Nielsen Norman Group](https://www.nngroup.com/articles/modal-nonmodal-dialog/)
Artikel waarin wordt beschreven wanneer je Modal Dialogs wel en niet hoort te gebruiken.

[The Current State of Homepage UX – 8 Common Pitfalls & Best Practices - Baymard Institute](https://arc.net/l/quote/uvpyqgph)
Artikel waarin het inzicht wordt gedeeld dat Modal Dialogs op de homepagina door enkele respondenten als spam wordt ervaren.

[Modals and Pop-ups design best practices (with examples) - Youtube](https://www.youtube.com/watch?v=-I5ANqGyO4A)
Video waarin Vitaly Friedman best practices van Modal Dialogs (en pop-ups) toont.

[modalzmodalzmodalz](https://modalzmodalzmodalz.com/)
Single page website waarin het te overmatig gebruik van Modals Dialogs wordt aangekaart. Aangevuld met alternatieve interactie patronen en best practices voor als je toch besluit een Modal Dialog in te zetten.

[Where to Put Focus When Opening a Modal Dialog](https://adrianroselli.com/2025/06/where-to-put-focus-when-opening-a-modal-dialog.html)
Adrian Roselli beschrijft de verschillende afwegingen die je kan maken om te besluiten waar je de focus naar toe brengt bij het openen van een Modal Dialog.

[Bekijk het Button group component voor meer informatie](https://github.com/orgs/nl-design-system/discussions/234)

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://amsterdam.github.io/design-system/?path=/docs/components-containers-dialog--docs)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/react/src/Dialog)
- [Figma](https://www.figma.com/file/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?type=design&node-id=5786-12616&mode=design&t=wOxtYbR0BkF2hDyL-4)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-04-05 om 15 32 54](https://github.com/nl-design-system/backlog/assets/545207/d96056fe-f001-4227-9f02-f1fb91c4a6b5)

Small viewport
![Scherm­afbeelding 2024-04-05 om 15 45 29](https://github.com/nl-design-system/backlog/assets/545207/fa009353-0c28-43d8-8919-fa062a93d5a1)

With Scrollbar
Content taller than the dialog itself will scroll.
![Scherm­afbeelding 2024-04-05 om 15 43 06](https://github.com/nl-design-system/backlog/assets/545207/642c7c3b-54cf-4da0-989c-2b5c557f9660)

**Figma**
![Scherm­afbeelding 2024-04-05 om 15 45 01](https://github.com/nl-design-system/backlog/assets/545207/27a3d611-cc97-4e01-a03c-9506c739fbf6)

---

### Notities
- Verschil tussen Figma en Storybook wat betreft gedrag Button group bij small viewport.
- Richtlijnen en beschrijving van keyboard navigatie beschikbaar (Storybook).

[Bekijk het Button group component van Amsterdam voor meer informatie](https://github.com/orgs/nl-design-system/discussions/234#discussioncomment-8900448)

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-modal--docs)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/Modal)
- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=221-32&mode=design&t=nSbHIZsilITuEfNP-4)
- [Figma - WIP](https://www.figma.com/file/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?type=design&node-id=3420-155509&mode=design&t=VDfpOD6L2Q0RcZuF-4)

---

### Screenshots / Varianten

**Storybook**
Default
![Scherm­afbeelding 2024-04-05 om 16 38 19](https://github.com/nl-design-system/backlog/assets/545207/2ac99c69-7461-4ac5-806c-3b5646dae146)

Small viewport
![Scherm­afbeelding 2024-04-05 om 16 58 45](https://github.com/nl-design-system/backlog/assets/545207/7769d3c9-ff6d-4641-a90d-92882ac0a48c)

Scrollabe
![Scherm­afbeelding 2024-04-05 om 16 43 35](https://github.com/nl-design-system/backlog/assets/545207/0f5e82f0-633f-4a52-b249-b9edb25aae32)

**Figma Huidig**

Default
![Scherm­afbeelding 2024-04-05 om 16 56 11](https://github.com/nl-design-system/backlog/assets/545207/9b60a56b-3c3b-4bcc-b6ae-8005c9c5869f)

Small viewport
![Scherm­afbeelding 2024-04-05 om 16 56 21](https://github.com/nl-design-system/backlog/assets/545207/1dabadda-32d2-409d-8d04-213946c16fe7)

Zonder acties
![Scherm­afbeelding 2024-04-05 om 16 56 36](https://github.com/nl-design-system/backlog/assets/545207/a2e37dad-7320-44f2-8b5a-df4a69bb3a7d)

**Figma WIP**

![Scherm­afbeelding 2024-04-08 om 16 16 53](https://github.com/nl-design-system/backlog/assets/545207/26175538-ed99-4700-8fed-1693bf7f9661)


---

### Notities
-  Richtlijnen beschikbaar.
- Verschil tussen Figma (Huidig + WIP) en Storybook wat betreft gedrag Heading kleur en Button group bij small viewport.
- Modal dialog in Figma - WIP gelijk aan Utrecht (Figma).

[Bekijk het Button group component van Den Haag voor meer informatie](https://github.com/orgs/nl-design-system/discussions/234#discussioncomment-8900745)

**Comment by jeffreylauwers** [comment]
## DSO Toolkit

- [Documentatie website](https://dso-toolkit.nl/62.8.4/components/on-page-interactives/modal)
- [GitHub](https://github.com/dso-toolkit/dso-toolkit/tree/master/storybook/src/components/modal)

---

### Screenshots / Varianten

**Storybook**

Confirm
![Scherm­afbeelding 2024-04-08 om 08 46 22](https://github.com/nl-design-system/backlog/assets/545207/eb121aad-51ca-4c7f-93c1-68d6a8bdb386)

Small viewport

![Scherm­afbeelding 2024-04-08 om 08 49 49](https://github.com/nl-design-system/backlog/assets/545207/28862479-65c5-4a2f-ad05-4260f40d44f1)

Passive
![Scherm­afbeelding 2024-04-08 om 08 46 14](https://github.com/nl-design-system/backlog/assets/545207/e16e498c-adf2-4187-82e8-e63d575a6805)

Active
![Scherm­afbeelding 2024-04-08 om 08 46 31](https://github.com/nl-design-system/backlog/assets/545207/720744be-991b-45be-b86b-6dae8a8381d0)

Loading
![Scherm­afbeelding 2024-04-08 om 08 46 38](https://github.com/nl-design-system/backlog/assets/545207/cc51030d-fb26-4340-8a50-f6da58e3d49a)

With Datepicker
![Scherm­afbeelding 2024-04-08 om 08 46 46](https://github.com/nl-design-system/backlog/assets/545207/c6674ea5-746b-49f5-8217-19451c3b4388)

Fullscreen
![Scherm­afbeelding 2024-04-08 om 08 46 54](https://github.com/nl-design-system/backlog/assets/545207/0c44ea36-03fd-4e7d-84df-f767c85f06d9)

---

### Notities
- Richtlijnen beschikbaar. Inclusief richtlijnen over toegankelijkheid en role (dialog vs. alert dialog).
- Veel verschillende vormen in Storybook.
- Niet duidelijk over en ook een backdrop aanwezig is.

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/modal)

---

### Screenshots / Varianten

**Documentatie website**

Modal met backdrop
Deze responsive modal wordt doorgaans het meest gebruikt voor extra informatie en korte gebruikersacties. De backdrop zorgt ervoor dat de focus op de modal komt te liggen. De gebruiker wordt niet afgeleid door informatie achter de modal.

![Scherm­afbeelding 2024-04-08 om 08 57 22](https://github.com/nl-design-system/backlog/assets/545207/9c7aa953-92f7-4c61-ba7e-45c4ca240253)

![Scherm­afbeelding 2024-04-08 om 08 57 37](https://github.com/nl-design-system/backlog/assets/545207/76b846c1-2003-4a9a-87b0-974bd1b867ab)

Modal voor afbeeldingen
Deze modal laat een afbeelding vergroot zien.
![Scherm­afbeelding 2024-04-08 om 08 57 51](https://github.com/nl-design-system/backlog/assets/545207/499305ba-9f73-4522-8a49-cd313bb5ffa3)

XL modal
![Scherm­afbeelding 2024-04-08 om 08 58 00](https://github.com/nl-design-system/backlog/assets/545207/770f251b-d38a-46fe-aa4e-312406c9a4bc)

Large modal
![Scherm­afbeelding 2024-04-08 om 08 58 14](https://github.com/nl-design-system/backlog/assets/545207/2d7fc0d9-2b2c-463f-b2c1-7554b4c7d97c)

Small modal
![Scherm­afbeelding 2024-04-08 om 08 58 23](https://github.com/nl-design-system/backlog/assets/545207/d15f7e3b-916c-4470-9289-35dc3e2af648)

---

### Notities
- Richtlijnen beschikbaar.
- Naast breedte ook classes voor hoogte beschikbaar.
- Button in principe links uitgelijnd, maar lijkt rechts gepositioneerd te worden wanneer footer tekst bevat.
- Sluit button met label.
- Responsive gedrag lijkt te zijn dan bij een small viewport de modal de gehele viewport afdekt, maar kon dit niet heel goed checken.

**Comment by jeffreylauwers** [comment]
## Open Formulieren

[Storybook](https://open-formulieren.github.io/open-forms-sdk/?path=/story/private-api-formstepsavemodal--form-step-save-modal)

---

### Screenshots / Varianten

**Storybook**

![Scherm­afbeelding 2024-04-08 om 09 24 16](https://github.com/nl-design-system/backlog/assets/545207/b6065dd9-af0f-4eba-a5af-3a935ddb02cb)

Small viewport

![Scherm­afbeelding 2024-04-08 om 09 26 03](https://github.com/nl-design-system/backlog/assets/545207/18bbf759-2c3f-4ca3-b649-35547b80967a)

![Scherm­afbeelding 2024-04-08 om 09 26 09](https://github.com/nl-design-system/backlog/assets/545207/2f7e1813-b471-42e6-a34d-870b0376c62b)

---

### Notities
- Niet duidelijk of er ook een backdrop aanwezig is.

**Comment by jeffreylauwers** [comment]
## Rotterdam

- [GitHub](https://github.com/nl-design-system/rotterdam/tree/main/rotterdam-nlds-parent-wicket/docs-wicket/src/main/java/nl/rotterdam/nl_design_system/docs/wicket/dialog)
- [Figma](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=16268-10877&mode=design&t=ZoRRQKdHcB9RgeLT-4)

---

### Screenshots / Varianten

**Figma**

Small
Mostly used for text only, like an explanation or a notification after a certain action.
![Scherm­afbeelding 2024-04-08 om 09 44 09](https://github.com/nl-design-system/backlog/assets/545207/e9ee4156-1bae-4164-90cb-20527e41028b)

Medium
Used when some additional UI elements are required (e.g. tables).
![Scherm­afbeelding 2024-04-08 om 09 44 18](https://github.com/nl-design-system/backlog/assets/545207/867fcb4f-75b9-4016-bf9a-c59735b27cdf)

Large
Only used for heavy duty applications for employees.
![Scherm­afbeelding 2024-04-08 om 09 44 29](https://github.com/nl-design-system/backlog/assets/545207/bdd50a1c-58e5-4da1-b5b3-565eb98b7cbb)

---

### Notities
- In 3 formaten, documentatie per formaat beschikbaar.

[Bekijk het Button group component van Rotterdam voor meer informatie](https://github.com/orgs/nl-design-system/discussions/234#discussioncomment-8901169)

**Comment by jeffreylauwers** [comment]
## RVO

- [Documentatie website](https://nl-design-system.github.io/rvo/docs/components/modal/)
- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-dialog--documentatie)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/dialog)
- [Figma](https://www.figma.com/file/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?type=design&node-id=1113-17884&mode=design&t=9Hp1izYOhOlI2yLM-4)

---

### Screenshots / Varianten

**Figma**
![Scherm­afbeelding 2024-04-08 om 09 43 57](https://github.com/nl-design-system/backlog/assets/545207/1531442f-efa2-46e3-a073-6eaa9f177eb3)

---

### Notities
- Documentatie beschikbaar.
- Component lijkt enkel in Figma beschikbaar.
- Modal dialog lijkt naast sluit optie geen acties ([Button group](https://github.com/orgs/nl-design-system/discussions/234#discussioncomment-8901297)) te bevatten.

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-alert-dialog--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/alert-dialog)
- [Figma](https://www.figma.com/file/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?type=design&node-id=3113-16705&mode=design&t=29ziuDiu63oymNqw-4)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-04-08 om 16 05 09](https://github.com/nl-design-system/backlog/assets/545207/77f5d456-f7c6-4ae6-a92d-1bb14517059f)

Small viewport
![Scherm­afbeelding 2024-04-08 om 16 15 47](https://github.com/nl-design-system/backlog/assets/545207/ece083cd-eb59-4d3c-a509-23af6ab683c9)

Info
![Scherm­afbeelding 2024-04-08 om 16 05 15](https://github.com/nl-design-system/backlog/assets/545207/48d024ab-f1b7-4faa-bb18-398b14887633)

Warning
![Scherm­afbeelding 2024-04-08 om 16 05 21](https://github.com/nl-design-system/backlog/assets/545207/6de74a1f-c3ca-430a-a310-8ff1b1aca15b)

Error
![Scherm­afbeelding 2024-04-08 om 16 05 28](https://github.com/nl-design-system/backlog/assets/545207/9780c752-a708-4158-adca-7933402a0fb8)

**Figma**

![Scherm­afbeelding 2024-04-08 om 16 16 41](https://github.com/nl-design-system/backlog/assets/545207/bd68c356-a9a5-4479-988f-b50a2f211934)


---

### Notities
**Storybook**
- Genaamd: Alert dialog
- Lijkt een combinatie van [Alert](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-alert--docs) en [Backdrop](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-backdrop--docs) component.

**Figma**
- Gelijk aan Den Haag - WIP (Figma).

[Bekijk het Button group component van Utrecht voor meer informatie](https://github.com/orgs/nl-design-system/discussions/234#discussioncomment-8901449)

**Comment by hidde** [comment]
Ik vind het belangrijk om “modal” te zien als eigenschap van “dialog”, wat we doen door het component “modal dialog” te noemen, we vermijden zo alleen “modal” te gebruiken als naam. Als counterpart kunnen we dan los “non-modal dialog” introduceren? Met bij beide duidelijke uitleg, misschien in de richting van een beslisboom, over wanneer je het beste kunt kiezen voor een modal vs een non-modal dialog.

Dit sluit aan bij hoe modals op het web werken: een `dialog` element in HTML kan met `show()` als non-modal worden getoond en met `showModal()` als modal.  Non-modal dialogs worden soms ook popovers genoemd, en er is het `popover` attribuut in HTML dat hier specifiek voor is gemaakt, maar “non-modal dialog” is waarschijnlijk een makkelijkere naam, vanwege het contrast met “modal dialog”.

**Comment by rianrietveld** [comment]
Discussie tussen onderzoekers over de role="document"  in een modal.
Role="document" aan een pop-up toevoegen: https://github.com/WCAG-Audit-Discussions/NL-BE/issues/74

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Modal dialog op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=modal+dialog+%2366)

**Comment by Rozerinay** [comment]
## Aanpassing Icon positie in Toolbar Button van Modal Dialog

Bij de Toolbar Button in de Modal Dialog gaf Marieke aan graag de volgorde van tekst en icoon te willen omdraaien, van X Sluiten naar Sluiten X. Er was op dat moment alleen ondersteuning voor een icoon aan het begin (icon-start), niet aan het einde (icon-end).

Jeffrey gaf aan dat deze component qua spacing gelijk is aan de algemene Button, en dat het technisch geen probleem is om een icon-end optie toe te voegen. Deze optie is inmiddels toegevoegd. 

Bron: uitwisseling tussen Marieke en @jeffreylauwers  in de NL Design System Figma bibliotheek.

**Comment by eslook** [comment]
**Gerelateerd aan de Modal Dialog: tijdens de estafettemodeldag van vrijdag 28 november hebben we een werksessie gedaan over de Alert Dialog.**

Het eindresultaat hiervan is: we gaan **door** met Alert Dialog in Help Wanted. Geïnteresseerd in de hoe en wat? Je kan [in de Alert Dialog GitHub discussie](https://github.com/orgs/nl-design-system/discussions/302#discussioncomment-15136732) de samenvatting vinden van de werksessie, plus een link naar de FigJam! Meningen, inzichten, input? Die kan je gewoon reageren in de GitHub discussie!

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Modal Dialog op het Candidate bord](https://github.com/orgs/nl-design-system/projects/80/views/1?filterQuery=Modal+dialog)
